java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4950,4 +4951,92 @@ public void sintercard(BaseClient client) {
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String key3 = "{key}-3" + UUID.randomUUID();
+        String key4 = "{key}-4" + UUID.randomUUID();
+        String foobarString = "foobar";
+        String helloString = "hello";
+        String[] keys = new String[] {key1, key2, key3};
+        Transaction setFoobarTransaction = new Transaction();
+        Transaction setHelloTransaction = new Transaction();
+        String[] expectedExecResponse = new String[] {OK, OK, OK};
+
+        // Returns null when a watched key is modified before it is executed in a transaction command.
+        // Transaction commands are not performed.
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertEquals(OK, standaloneClient.set(key2, helloString).get());
+        setFoobarTransaction.set(key1, foobarString).set(key2, foobarString).set(key3, foobarString);
+        assertEquals(null, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(null, standaloneClient.get(key1).get());
+        assertEquals(helloString, standaloneClient.get(key2).get());
+        assertEquals(null, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with a read command on the watch key before
+        // transaction is executed.
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertEquals(helloString, standaloneClient.get(key2).get());
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(foobarString, standaloneClient.get(key1).get());
+        assertEquals(foobarString, standaloneClient.get(key2).get());
+        assertEquals(foobarString, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with an unmodified watched key
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(foobarString, standaloneClient.get(key1).get());
+        assertEquals(foobarString, standaloneClient.get(key2).get());
+        assertEquals(foobarString, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with a modified watched key but is not in the
+        // transaction.
+        assertEquals(OK, standaloneClient.watch(new String[] {key4}).get());
+        setHelloTransaction.set(key1, helloString).set(key2, helloString).set(key3, helloString);
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setHelloTransaction).get());
+        assertEquals(helloString, standaloneClient.get(key1).get());
+        assertEquals(helloString, standaloneClient.get(key2).get());
+        assertEquals(helloString, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with an unmodified watched key
+        assertEquals(OK, standaloneClient.watch(keys).get());
```

Comment:
Isn't this block equivalent to the one two blocks above? Can this be removed or do we need to test the same thing twice for some reason?

Reply:
Good catch. Thanks